apps/login/app/(login)/login/route.ts

```diff
@@ -30,47 +32,144 @@ function findSession(
       (s) => s.factors?.user?.loginName === authRequest.loginHint
     );
   }
+  if (sessions.length) {
+    return sessions[0];
+  }
   return undefined;
 }
-
 export async function GET(request: NextRequest) {
   const searchParams = request.nextUrl.searchParams;
   const authRequestId = searchParams.get("authRequest");
+  const sessionId = searchParams.get("sessionId");
+
+  const sessionCookies: SessionCookie[] = await getAllSessions();
+  const ids = sessionCookies.map((s) => s.id);
+  let sessions: Session[] = [];
+  if (ids && ids.length) {
+    sessions = await loadSessions(ids);
+  }
+
+  if (authRequestId && sessionId) {
+    console.log(
+      `Login with session: ${sessionId} and authRequest: ${authRequestId}`
+    );
+
+    let selectedSession = sessions.find((s) => s.id === sessionId);
+
+    if (selectedSession && selectedSession.id) {
+      console.log(`Found session ${selectedSession.id}`);
+      const cookie = sessionCookies.find(
+        (cookie) => cookie.id === selectedSession?.id
+      );
+
+      if (cookie && cookie.id && cookie.token) {
+        console.log(`Found sessioncookie ${cookie.id}`);
+
+        const session = {
+          sessionId: cookie?.id,
+          sessionToken: cookie?.token,
+        };
+
+        const { callbackUrl } = await createCallback(server, {
+          authRequestId,
+          session,
+        });
+        return NextResponse.redirect(callbackUrl);
+      }
+    }
+  }
 
   if (authRequestId) {
+    console.log(`Login with authRequest: ${authRequestId}`);
     const { authRequest } = await getAuthRequest(server, { authRequestId });
-    const sessionCookies: SessionCookie[] = await getAllSessions();
-    const ids = sessionCookies.map((s) => s.id);
+    let organization;
 
-    let sessions: Session[] = [];
-    if (ids && ids.length) {
-      sessions = await loadSessions(ids);
-    } else {
-      console.info("No session cookie found.");
-      sessions = [];
+    if (
+      authRequest?.scope &&
+      authRequest.scope.find((s) => ORG_SCOPE_REGEX.test(s))
+    ) {
+      const orgId = authRequest.scope.find((s) => ORG_SCOPE_REGEX.test(s));
+
+      if (orgId) {
+        const matched = orgId.replace("urn:zitadel:iam:org:id:", "");
```

Comment:
let's match and extract the id using a regex group?

Reply:
done that now